### PR TITLE
Mhs/cumulus 2216/make statistics inclusion optional on the collections endpoint

### PR DESCRIPTION
**Pull request by flamingbear**
**Summary:**

Splits es/Collection into BaseCollection and Collection and leaves the granule statistics aggregation in Collection, while moving the base functionality into BaseCollection.

/collection and /collection/active endpoints now return collections without aggregate statistics by default. To get the original behavior a query param of `includeStats=true` must be present on the request

Addresses [CUMULUS-2226: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2216)

## Changes
* Collection endpoint looks for an `includeStats` param and chooses which type of collection to instantiate before performing the endpoint's search.

## PR Checklist

- [X] Update CHANGELOG
- [X] Unit tests
- [X] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

**Comment by jennyhliu** [comment]
The `Example request` for `List collections` and `List collections with active granules` doesn't have parameter `includeStats=true`, but  the `Example Response` includes `stats`, this could cause confusion

File: packages/api/endpoints/collections.js [hunk 1]
```diff
@@ -27,14 +28,16 @@ const log = new Logger({ sender: '@cumulus/api/collections' });
  * @returns {Promise<Object>} the promise of express response object
  */
 async function list(req, res) {
-  const { getMMT, ...queryStringParameters } = req.query;
-  const collection = new Collection(
+  const { getMMT, includeStats, ...queryStringParameters } = req.query;
+  const CollectionClass = includeStats === 'true' ? Collection : BaseCollection;
```
Comment (jennyhliu) [inline packages/api/endpoints/collections.js:32]: Can we pass `includeStats` to `Collection` class, and let the class decide if the stats should be included or not?

Comment (flamingbear) [reply packages/api/endpoints/collections.js:32]: I could do that. I actually thought this was cleaner. But I'm fine changing it if you think that is a better way to do it.

Comment (jennyhliu) [reply packages/api/endpoints/collections.js:32]: I prefer letting `Collection` class worry about the `includeStats` flag.

Comment (flamingbear) [reply packages/api/endpoints/collections.js:32]: Ok, I'm putting that up already.  Thanks.

**Comment by flamingbear** [comment]
closing in favor of #1954

**Pull request closed without merging**